app/lib/differential_expression_service.rb

```diff
@@ -185,6 +153,93 @@ def self.run_differential_expression_job(cluster_file, study, user, annotation_n
     end
   end
 
+  # find all eligible annotations for DE for a given study
+  #
+  # * *params*
+  #   - +study+        (Study) => Associated study object
+  #   - +machine_type+ (String) => Override default VM machine type
+  #   - +dry_run+      (Boolean) => Indication of whether or not this is a pre-flight check
+  #   - +skip_existing+   (Boolean) => Skip annotations that already have DE results
```

Comment:
I wonder if there's a good way to DRY up this sort of copy-and-paste docstring.  

Various motivations and ideas to do so are discussed at length over in e.g. https://github.com/matplotlib/matplotlib/issues/9414.  Alas, it seems no solutions were found (and it's a different programming language anyways), so just consider this food for thought.

Reply:
Yeah, that would be nice, but I think there's probably no good solution that will work with IDEs universally.